autoPVS1_from_VEP_vcf.py

```diff
@@ -1,48 +1,131 @@
+                summary['pick'] +=1
+                tie_break_return =  pick[0], pick[1], summary
+                pick_break = 1
+                break
+        # If not found, warn that first rank hit used
+        if not pick_break:
+            print("WARN: No matching PICK, default to longest transcript", file=sys.stderr)
+            summary['length'] +=1
+            tie_break_return =  tx_candidates[tx_len_dict[longest]], candidate_csqs[tx_len_dict[longest]], summary
+    else:
+        summary['rank'] += 1
+        tie_break_return = tx_candidates[champ], candidate_csqs[champ], summary
+    return tie_break_return
+
+
+def get_candidates(record_csq, csq_fields):
+    """
+    Takes in a pysam csq record and the csq_fields list parsed from the record description.
+    Returns a list of picked csqs, matching transcript objects, and the PICK csq
+    1. See if the tx candidate is canonical
+    2. If canonical, check to see if it exists in the tool transcript reference
+    3. Record the PICK value for rule 5 from pick_transcript
+    """
+    canon_idx = csq_fields.index("CANONICAL")
+    pick_index = csq_fields.index("PICK")
+    refseq_index = csq_fields.index('RefSeq')
+    pick_value = "1"
+    canon_value = "YES"
+    
+    candidate_csqs = []
+    tx_candidates = []
+
+    for csq in record_csq:
+        info_list = csq.split("|")
+        refseq_id = info_list[refseq_index]
+        if info_list[pick_index] == pick_value:
+            pick = (get_transcript(refseq_id, transcripts_hg38), info_list)
+        if info_list[canon_idx] == canon_value and refseq_id != "":
+            check = get_transcript(refseq_id, transcripts_hg38)
+            if check is not None:
+                candidate_csqs.append(info_list)
+                tx_candidates.append(check)
+    return candidate_csqs, tx_candidates, pick
+
+
+def pick_transcript(record, csq_fields, summary):
+    """
+    Function to process a record and choose a representative transcript
+    Does so by doing the following:
+    1. See if the tx candidate is canonical
+    2. If canonical, check to see if it exists in the tool transcript reference
+    3. If multiple hits exists, use rank list to chose the worst
+    4. If still multiple (meaning ranks are tied), use longest transcript length as defined by the reference
+    5. If no hits from step 1 and 2, just revert to using PICK
+    """
+    candidate_csqs, tx_candidates, pick = get_candidates(record.info['CSQ'], csq_fields)
+    if len(candidate_csqs) == 1:
+        summary['canonical'] += 1
+        return_values =  tx_candidates[0], candidate_csqs[0], summary
+
+    # tie break with rank - if multiple share rank, go with PICK, if not go with tx length
+    if len(candidate_csqs) > 1:
+        return_values = tie_breaker(record, candidate_csqs, tx_candidates, pick, csq_fields, summary)
+    # go with PICK
+    if len(candidate_csqs) == 0:
+        summary['pick'] +=1
+        return_values =  pick[0], pick[1], summary
```

Comment:
You can only go down a single path. Using `if` for each would be appropriate if each were a possibility.
```suggestion
    candidate_csqs, tx_candidates, pick = get_candidates(record.info['CSQ'], csq_fields)
    # go with PICK
    if len(candidate_csqs) == 0:
        summary['pick'] +=1
        return_values =  pick[0], pick[1], summary
    elif len(candidate_csqs) == 1:
        summary['canonical'] += 1
        return_values =  tx_candidates[0], candidate_csqs[0], summary
    # tie break with rank - if multiple share rank, go with PICK, if not go with tx length
    else:
        return_values = tie_breaker(record, candidate_csqs, tx_candidates, pick, csq_fields, summary)
```